### PR TITLE
Investiate the use of graph_tool library and possibility of its removal

### LINKED ISSUE #884: Investiate the use of graphql library and possibility of its removal
We have two solvers, I think only the allsolutions solver is using it and we are not really using this solver anymore. If possible create a brach that removes this solver altogether

### PR DESCRIPTION
**Pull request by kristinagalik**
Closes #884

### CODE REVIEW

File: txmatching/solvers/ilp_solver/result.py [hunk 1]
```diff
@@ -2,7 +2,7 @@
 
 from txmatching.solvers.ilp_solver.solution import Solution, Status
 
-
+# this is not used anywhere, do I keep it for future possibilities or not
```
Comment (kristinagalik) [inline txmatching/solvers/ilp_solver/result.py:5]: @kubantjan

File: txmatching/solvers/ilp_solver/logging_utils.py [hunk 2]
```diff
@@ -1,3 +1,4 @@
+# this is not used anywhere too
```
Comment (kristinagalik) [inline txmatching/solvers/ilp_solver/logging_utils.py:1]: @kubantjan

**Comment by kristinagalik** [comment]
@kubantjan 

* deleted AllSolutionSolver
* things that were used in both solvers were moved from AllSolutionSolver to ILPSolver
* functions and classes from ILPSolver arent used anywhere else in the code (apart from sensible ones, `solver_base` and `solver_from_config`)
* only essential things are imported from the rest of the code into the `ilp_solver` files, and thats too only into `txm_configuration_for_ilp` and `solve_ilp`
* there are 2 classes that arent used anywhere, should i delete them or do they seem like we could use them in the future?

**Review by kubantjan — CHANGES_REQUESTED** [review]
1) Klidne odeber i ty nepouzity knihovny.
2) myslim si, ze takto to odebrat tu zavislost nestaci, ze je jeste potreba ji dat pryc z conda.yml -> to je to dulezity misto, kde se vyskytuji dependencies.
3) to bude asi chtit pak updatnout base image (neco je v readme, pokud si nebudes vedet rady, rekni si)
4) to by melo zajistit, ze uz to beha jak  na windows a M1 -> updatnout readme (zkontrolvoat, jestli jeste nejaky dalsi updates nejsou potreba)

**Comment by kristinagalik** [comment]
@kubantjan 
skusala som to rozbehat na windowse a aj tak treba wsl2 kvoli gunicornu

**Comment by kubantjan** [comment]
> @kubantjan skusala som to rozbehat na windowse a aj tak treba wsl2 kvoli gunicornu

Lets keep english everywhere :). I forget sometimes as well but we have to remind each other 

Ok but gunicorn is used just in production setup, when runnning locally we are running flask only. So this should not be a problem. The only issue is that we have gunicorn in even in the conda.yml that is meant for local run.

could you try to remove gunicorn from conda yml and run it locally?

if it does not help, we wont deal with it further but if it did it would be cool.

**Comment by kubantjan** [comment]
I will have couple more comments in the end, but lets do it via a call, I have sent you a message

**Comment by kubantjan** [comment]
1) check with @NotLubo  whether it works for him with graph tool only in docker or no
2) check whether graph tool is the best thing we can use (from this https://graph-tool.skewed.de/performance it seems that it is, but it is quite old)
3) do not delete the solver in the end, but maybe use some refactoring if it was reasonable
4) study the handbook of kepsoft and understand the inputs and outputs of the solver that is used there
5) select an option from the two below, depending on which one would be easier
a) create an issue to replace topology.max_cliques(path_intersections_graph) with ILP
b) create an issue that woud replace   topology.max_cliques(path_intersections_graph) with kepsoft optimizer

**Pull request closed without merging**